src/resolvers/login-resolver.ts

```diff
@@ -1,16 +1,48 @@
+import { CustomHttpError } from '../errors.js';
+import { prisma } from '../database.js';
+import * as bcrypt from 'bcrypt';
+import { EMAIL_NOT_FOUND_MESSAGE, INCORRECT_PASSWORD_MESSAGE } from '../utils/validators.js';
+
 export interface LoginInput {
   email: string;
   password: string;
 }
 
-export const loginResolver = (_, args: { data: LoginInput }) => {
+export const loginResolver = async (_, args: { data: LoginInput }) => {
+  const { email, password } = args.data;
+
+  const user = await findUserByEmail(email);
+  checkPassword(password, user.password);
+
   return {
     user: {
-      id: 13,
-      name: 'guilherme',
-      email: 'teste@taqtile.com.br',
-      birthDate: '25/04/2003',
+      id: user.id,
+      name: user.name,
+      email: user.email,
+      birthDate: user.birthDate,
     },
     token: 'mockToken',
   };
 };
+
+const findUserByEmail = async (email: string) => {
+  try {
+    const user = await prisma.user.findUniqueOrThrow({
+      where: {
+        email,
+      },
+    });
+    return user;
+  } catch {
+    throw new CustomHttpError(400, EMAIL_NOT_FOUND_MESSAGE);
+  }
+};
+
+const checkPassword = (password: string, encryptedPassword: string) => {
+  if (!isPasswordCorrect(password, encryptedPassword)) {
+    throw new CustomHttpError(400, INCORRECT_PASSWORD_MESSAGE);
+  }
+};
+
+const isPasswordCorrect = (password: string, encryptedPassword: string) =>
+  bcrypt.compareSync(password, encryptedPassword);
```

Comment:
Tem uma diferença sutil aqui, e que no seu contexto não vai fazer muita diferença, mas no geral já fica a recomendação de quando tiver a opção de usar uma função async, costuma ser uma melhor prática (isso quando tem essas opções de `fooSync`).

Se quiser mais detalhes sobre o motivo, pode ler [aqui](https://github.com/kelektiv/node.bcrypt.js#why-is-async-mode-recommended-over-sync-mode) com mais calma.

Reply:
Troquei para usar a função assíncrona